common/src/main/scala/db/SqlRegistrationRepository.scala

```diff
@@ -87,18 +87,19 @@ class SqlRegistrationRepository[F[_]: Async](xa: Transactor[F])
       )
       ++ fr"GROUP BY token, platform"
       )
-      .query[(String, String)]
+      .query[(String, String, String)]
       .stream
       .transact(xa)
-      .map{ case(token, platformString) => {
+      .map{ case (token, platformString, buildTierString) => {
         val maybePlatform = Platform.fromString(platformString)
         if(maybePlatform.isEmpty) {
           logger.error(s"Unknown platform in db $platformString")
         }
-        (token, maybePlatform)
+        val maybeBuildTier = BuildTier.fromString(buildTierString)
```

Comment:
So here you'll have `buildTierString` of type `Option[String]`, and a function `BuildTier.fromString` which takes a `String` and returns an Option[BuildTier].

How can you reconcile the two such that `maybeBuildTier` is of type `Option[BuildTier]` ?

If that's unclear I suggest you start a worksheet to work on that part

Reply:
So here's an example of setup where you could try to guess the missing part. Don't pay attention to what the code actually does (it's rubbish), but only focus on the types!


```scala

case class BuildTier(s: String)

def buildTierFromString(buildTierString: String): Option[BuildTier] = Some(buildTierString)

val myOptionalString: Option[String] = Some("Hello")

val maybeBuildTier: Option[BuildTier] = <insertSomethingHere>
```

Without touching anything else than what's in the angle brakets `<insertSomethingHere>` can you get `maybeBuildTier` to be the result of calling `buildTierFromString` with the content of `myOptionalString` such that the code compiles?